### ISSUE TITLE
Make network inference speed faster

### ISSUE THREAD

**Issue by BenBaek** [MaybeShewill-CV/lanenet-lane-detection#325]
Hi, 
I am a student studying deep learning.
First of all, thank you for your implementing.

I did some experiments by changing the base network from your code.
I trained through my own training dataset, and in my first trial I only change the base network and didn't change loss, global configuration.

Some Nan issues came up, and I modifed the discriminative loss error using clip-by-value.

Training was carried out normally and the tests were carried out when the loss was converged near a certain value.

For VGG model, the results of binary segmentation and instance segmentation were neat and lean.

Some similar results, but when replaced by MobileNet binary segmentation results are looks thick and representation of lane will be combined in vanishing point. In addition, as a result of instance segmentation, the output is also thick and unclear, sometimes leading to the same embedding for different lanes.

Output images are shown as below:
VGG - 
![vgg](https://user-images.githubusercontent.com/39553764/69938929-248bf200-1511-11ea-905b-a767f88b8e93.png)
MobileNet - 
![mobilenet](https://user-images.githubusercontent.com/39553764/69938939-29e93c80-1511-11ea-80b4-e4345e5ba4ba.png)

 I saw your code and the loss function organized with weight update through heuristic value.
Should I find a suitable weight value for my network model and training dataset ? or should a new loss function be configured for troubleshooting?

Would it be helpful to increase the predictive dimension in instance loss, or adjust the delta distance and variance value?

I would like to know how did you specified the certain values.

**Comment by MaybeShewill-CV** [comment]
@BenBaek If you want to get a more lean and neat binary segmentation result. You may simply test enlarge your input image size:)

**Comment by BenBaek** [comment]
> @BenBaek If you want to get a more lean and neat binary segmentation result. You may simply test enlarging your input image size:)

Thanks for your reply,
I have already trained both 'VGG' and 'MobileNet' for 256x128 and 512x256 inputs.
VGG with 256x128 input shows more clear results than MobileNet with 512x256 input.

VGG instance loss converges to near 0.2, and MobileNet instance loss converges to near 2.0, so
I suspect that this phenomenon is mired in the local minima problem.

How do you think about this problem?
Should I change some parameter about loss function?

**Comment by MaybeShewill-CV** [comment]
@BenBaek What I mean is that if you train the model with input size [256, 512] you may test it with input size [512, 1024] you may get lean and neat binary segmentation result. As for the loss problem you mentioned above that's mainly caused by the different feature extractor performance by two different base net. I think you need to adjust some params in instance loss function:)

**Comment by BenBaek** [comment]
> @BenBaek What I mean is that if you train the model with input size [256, 512] you may test it with input size [512, 1024] you may get lean and neat binary segmentation result. As for the loss problem you mentioned above that's mainly caused by the different feature extractor performance by two different base net. I think you need to adjust some params in instance loss function:)

I'll try to experiment and find the proper parameters as you say.
Thanks!

**Comment by MaybeShewill-CV** [comment]
@BenBaek Wish you good luck and welcome to share your result with us if you're willing to:)
